packages/graphiql-react/src/history/context.tsx

```diff
@@ -76,51 +94,45 @@ export function HistoryContextProvider(props: HistoryContextProviderProps) {
   const [items, setItems] = useState(historyStore.current?.queries || []);
 
   const addToHistory: HistoryContextType['addToHistory'] = useCallback(
-    ({ query, variables, headers, operationName }) => {
-      historyStore.current?.updateHistory(
-        query,
-        variables,
-        headers,
-        operationName,
-      );
+    (operation: QueryStoreItem) => {
+      historyStore.current?.updateHistory(operation);
       setItems(historyStore.current.queries);
     },
     [],
   );
 
   const editLabel: HistoryContextType['editLabel'] = useCallback(
-    ({ query, variables, headers, operationName, label, favorite }) => {
-      historyStore.current.editLabel(
-        query,
-        variables,
-        headers,
-        operationName,
-        label,
-        favorite,
-      );
+    (operation: QueryStoreItem) => {
+      historyStore.current.editLabel(operation);
       setItems(historyStore.current.queries);
     },
     [],
   );
 
   const toggleFavorite: HistoryContextType['toggleFavorite'] = useCallback(
-    ({ query, variables, headers, operationName, label, favorite }) => {
-      historyStore.current.toggleFavorite(
-        query,
-        variables,
-        headers,
-        operationName,
-        label,
-        favorite,
-      );
+    (operation: QueryStoreItem) => {
+      historyStore.current.toggleFavorite(operation);
       setItems(historyStore.current.queries);
     },
     [],
   );
 
+  const setActive: HistoryContextType['setActive'] = useCallback(
+    (item: QueryStoreItem) => {
+      return item;
+    },
+    [],
+  );
```

Comment:
what's purpose of this function?

Reply:
I'm adjusting things in my UI based on the active item so this helps to know when the active item is changed.

Reply:
This PR has changed a bit since I first started it though... I can play around and see if I can it to work without this function this afternoon.

Reply:
Hmm... couldn't figure out how to get it to work without the setActive. I'm using the <HistoryContext.Provider value={history}> and implementing my own history... addToHistory, editLabel, etc, storing the items in a backend instead of in localStorage, and resetting the results pane when I click on one of those items in history based on the id it's stored in the backend because results can change over time for the same operation.